File: 2316-count-hills-and-valleys-in-an-array/2316-count-hills-and-valleys-in-an-array.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        res = 0
        for i in range(1, len(nums) - 1):
            b1 = i - 1
            b2 = i + 1
            while i + 1 < len(nums) and nums[i] == nums[i + 1]:
                b2 += 1
                i += 1
            if b2 < len(nums):
                if (nums[i] > nums[b1] and nums[i] > nums[b2]) or (
                    nums[i] < nums[b1] and nums[i] < nums[b2]
                ):
                    res += 1
        return res

File: 2316-count-hills-and-valleys-in-an-array/2316-count-hills-and-valleys-in-an-array.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        int res = 0;
        for (int i = 1; i < nums.size() - 1; i++) {
            int border1 = i - 1;
            int border2 = i + 1;
            while (i + 1 < nums.size() && nums[i] == nums[i + 1]) {
                border2++;
                i++;
            }
            if (border2 < nums.size()) {
                if ((nums[i] > nums[border1] && nums[i] > nums[border2]) ||
                    (nums[i] < nums[border1] && nums[i] < nums[border2])) {
                    res++;
                }
            }
        }
        return res;
    }
};